### PR TITLE
readme: organize examples by category

### PR DESCRIPTION
**Pull request by danicat**
I've made the modifications proposed by #309. I'm not sure if this is the best layout, because some examples seem to fit to more than one category.

I've added a `Getting Started` category to the proposal because I couldn't fit those samples to a single category. Also, there are no Networking examples available right now, but I've left the title there as a placeholder for when those are available.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,22 +67,32 @@ Note: samples under `docs/appengine` are not shown because they mostly do not ru
...
-|[monitoring/listresources](monitoring/listresources)|Command listresources lists the Google Cloud Monitoring v3 Environment against an authenticated user.|
+
+### Storage
+
+|Path|Description|
+|---|---|
+|[datastore/tasks](datastore/tasks)|A simple command-line task list manager to demonstrate using the cloud.google.com/go/datastore package.|
+|[storage/buckets](storage/buckets)|Sample buckets creates a bucket, lists buckets and deletes a bucket using the Google Storage API.|
+
+### Networking
```
Comment (broady) [inline README.md]: I think we can drop it.

File: README.md [hunk 2]
```diff
@@ -27,12 +27,31 @@ go list -f '|[{{.Dir}}]({{.Dir}})|{{.Doc}}|' ./... | egrep -v '/(internal|docs/a
...
+### Getting Started
+
+|Path|Description|
+|---|---|
+|[getting-started/bookshelf](getting-started/bookshelf)|Package bookshelf contains the bookshelf database and app configuration, shared by the main app module and the worker module.|
+|[getting-started/bookshelf/app](getting-started/bookshelf/app)|Sample bookshelf is a fully-featured app demonstrating several Google Cloud APIs, including Datastore, Cloud SQL, Cloud Storage.|
+|[getting-started/bookshelf/pubsub_worker](getting-started/bookshelf/pubsub_worker)|Sample pubsub_worker demonstrates the use of the Cloud Pub/Sub API to communicate between two modules.|
+|[getting-started/helloworld](getting-started/helloworld)|Sample helloworld is a basic App Engine flexible app.|
+
+### Management
```
Comment (broady) [inline README.md]: maybe this?

> Logging and Monitoring (Stackdriver)

File: README.md (hunk 1 above)
Comment (danicat) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (danicat) [reply README.md]: Done

**Comment by danicat** [comment]
@broady There is anything else you need me to do on this one?

**Comment by tbpg** [comment]
Travis failure is unrelated. I rebased and updated the messages a bit. Merging. Thank you!

**Pull request merged**